packages/data-stores/src/plans/selectors.ts

```diff
@@ -131,26 +132,28 @@ export const getPlanByPath = (
 
 	return select( STORE_KEY )
 		.getSupportedPlans( locale )
-		.find( ( plan ) => plan.periodAgnosticSlug === planProduct?.periodAgnosticSlug );
+		.find( ( plan ) => plan.periodAgnosticSlug === planProduct.periodAgnosticSlug );
 };
 
 export const getPlanProduct = (
 	_state: State,
-	periodAgnosticSlug: string | undefined,
-	billingPeriod: PlanProduct[ 'billingPeriod' ] | undefined
+	periodAgnosticSlug?: string,
+	billingPeriod?: PlanProduct[ 'billingPeriod' ]
```

Comment:
@alshakero  I guess same as https://github.com/Automattic/wp-calypso/pull/50492#discussion_r586259136?

Reply:
Reverted in https://github.com/Automattic/wp-calypso/pull/50666/commits/0a1c8ee0c6c223355b229d7c078a36f48f2d2649